framework/include/base/MooseObject.h

```diff
@@ -32,6 +82,8 @@ InputParameters validParams<MooseObject>();
  * Every object that can be built by the factory should be derived from this class.
  */
 class MooseObject :
+  public virtual BlockRestrictableHelper, // see BlockRestrictable.h
+  public virtual BoundaryRestrictableHelper, // see BoundaryRestrictable.h
```

Comment:
We've done such a good job of keeping these out of the framework. You'll probably have to convince us that this is the best design.